pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -294,7 +300,32 @@ func (m *manager) GetTopologyHints(pod *v1.Pod, container *v1.Container) map[str
 	// Garbage collect any stranded resources before providing TopologyHints
 	m.removeStaleState()
 	// Delegate to active policy
-	return m.policy.GetTopologyHints(m.state, pod, container)
+	hint := m.policy.GetTopologyHints(m.state, pod, container)
+	podAnnotations := pod.GetAnnotations()
+	if cmCPUPolicyName == PolicyStatic && hint != nil {
+		if podAnnotations == nil {
+			pod.SetAnnotations(map[string]string{allowCPUQuotaDisabled: "true"})
```

Comment:
i don't understand why are you are setting annotations on this pod object, nothing should be recorded on the pod resource itself.